### PR TITLE
Bookmark feature added

### LINKED ISSUE #46: Feature: repository bookmarking
### Description:
To improve user experience, users should have the ability to bookmark repositories they find interesting or plan to contribute to. This feature will allow users to easily access their favourite repositories later.

### Tips for the issue:
- Design and add a bookmark icon to repository cards.
- Create a "Bookmarks" section in the app.
- Implement the ability to add/remove repositories from bookmarks.
- Make sure bookmarked repos remain persistent in the bookmark tab

### To do:
- [x] Ask us to assign the issue
- [x] Once the issue is assigned, you can start working on it
- [x] Create a PR

### Resource:
- [Material 3 Design](https://m3.material.io/)
- [Flutter UI docs](https://docs.flutter.dev/ui)
- [Storing Data](https://docs.flutter.dev/cookbook/persistence)

### Notes:
The task is **assigned to contributors on a first-come, first-serve basis**, and the **contributor has to report progress every 3 days** to ensure active development on the tasks.

### PR DESCRIPTION
**Pull request by Yash-Waghmare**
### Changes made:
-   bookmarks bottom navigation added.
-   can add and remove the bookmark repositories

### Solution to problem:
This PR will allow the individual to add the repositories to their collection and also remove it from the collection, and the collection is persistent.

### Linked issues
Resolves #46

### CODE REVIEW

File: pubspec.yaml
Comment (dk-a-dev) [inline pubspec.yaml:1]: Line 44 and Line 39
> Error: Duplicate library

File: lib/screens/bookmark.dart
Comment (dk-a-dev) [inline lib/screens/bookmark.dart:1]: Line 64-68 remove the shape of appbar

Comment (Yash-Waghmare) [reply lib/screens/bookmark.dart:1]: Done

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
image
I am getting this error and repo is not bookmarking  Line 28 of new repo widget

**Comment by Yash-Waghmare** [comment]
> image I am getting this error and repo is not bookmarking Line 29 of new repo widget

It was working fine, Let me check once

**Comment by Yash-Waghmare** [comment]
> image I am getting this error and repo is not bookmarking Line 29 of new repo widget

Hey, I am not getting that error but I have made the description string nullable in boormaed_repo_model.dart,
Check it once 

image

**Comment by dk-a-dev** [comment]
@Yash-Waghmare conflicts generated will be resolved by me dw, its just basic refactoring

**Comment by dk-a-dev** [comment]
@Yash-Waghmare 
that issue is because if a desc from a repo is null or not provided then its giving that error so
remove null check ? from ur model
instead replace 
> final String repoDescription = widget.listData[widget.index]['description'].toString();
add .tostring() there
rest everything seems fine I will resolve the conflict and will merge this asap

**Comment by dk-a-dev** [comment]
@Yash-Waghmare also new issues will be added by tomorrow to this project

**Comment by Yash-Waghmare** [comment]
> Collaborator



> @Yash-Waghmare that issue is because if a desc from a repo is null or not provided then its giving that error so remove null check ? from ur model instead replace
> 
> > final String repoDescription = widget.listData[widget.index]['description'].toString();
> 
> add .tostring() there rest everything seems fine I will resolve the conflict and will merge this asap EXAMPLE CHECK - liwid-app( a repo with no desc)

Okay, I will check it once, and do the needfull

**Comment by Yash-Waghmare** [comment]
> > Collaborator
> 
> > @Yash-Waghmare that issue is because if a desc from a repo is null or not provided then its giving that error so remove null check ? from ur model instead replace
> > > final String repoDescription = widget.listData[widget.index]['description'].toString();
> > 
> > 
> > add .tostring() there rest everything seems fine I will resolve the conflict and will merge this asap EXAMPLE CHECK - liwid-app( a repo with no desc)
> 
> Okay, I will check it once, and do the needfull

Done

**Review by dk-a-dev — APPROVED** [review]
will be merged by tomorrow
NOTE:  do not resolve conflicts, I will do

**Comment by Yash-Waghmare** [comment]
> will be merged by tomorrow NOTE: do not resolve conflicts, I will do

great, Thanks !!

**Pull request merged**